network/network_node.py

```diff
@@ -0,0 +1,22 @@
+class NetworkNode:
+    def __init__(self, ip, mac, gateway_ip, gateway_mac, hostname, os=None):
+        self.ip = ip
+        self.mac = mac
+        self.gateway_ip = gateway_ip
+        self.gateway_mac = gateway_mac
+        self.hostname = hostname
```

Comment:
consider having all non-essential fields (os, hostname, etc) inside a dict called tags, so that nodes do not need to have them, and you can add an arbitrary number. You can then have some kind of add_tag function.